tls/s2n_early_data.c

```diff
@@ -82,3 +89,24 @@ int s2n_psk_set_context(struct s2n_psk *psk, const uint8_t *context, uint16_t si
     memcpy_check(context_blob->data, context, size);
     return S2N_SUCCESS;
 }
+
+S2N_RESULT s2n_early_data_config_clone(struct s2n_psk *new_psk, struct s2n_early_data_config *old_config)
+{
```

Comment:
Since the function is only cloning the `s2n_early_data_config`,  I think it would be better to pass only the configs (old and new) as the argument. 

```
S2N_RESULT s2n_early_data_config_clone(struct s2n_early_data_config *old_config, struct s2n_early_data_config *new_config);
```

Reply:
If we do that, then we can't use any of the public APIs to configure early data, because they all take psks because customers don't need to know that we model early data config separately. And it seems overly complicated to add non-public versions of all the public APIs just to operate on the early data config directly.